crates/rolldown_binding/src/bundler.rs

```diff
@@ -66,12 +67,18 @@ impl Bundler {
       worker_manager,
     )?;
 
-    Ok(Self {
+    let bundler = Self {
       cwd: ret.bundler_options.cwd.clone().unwrap_or_else(|| std::env::current_dir().unwrap()),
       inner: Arc::new(Mutex::new(NativeBundler::with_plugins(ret.bundler_options, ret.plugins))),
       log_level,
       on_log,
-    })
+    };
+
+    if !ret.warnings.is_empty() {
+      block_on(bundler.handle_warnings(ret.warnings));
```

Comment:
You don't need to handle warnings. Just make it exist in the final `BundleOutput#warnings`. It gets printed automatically by handlers in `rolldown_binding`/`integrstion_test.rs`.

Reply:
We cannot store `warnings` within `Bundler::new`. Should we consider adding a `warnings` field to the struct `Bundler` or `NativeBundler`?

Reply:
Yeah. Adding a `Bundler#warnings` would be fine.

Reply:
I eventually added it to `NativeBundler` because its methods are mutable and common for both `rolldown_binding` and `integration_test.rs`.